Get-DellWarranty.psm1

```diff
@@ -0,0 +1,95 @@
+        -- Fully qualified domain name (FQDN)\computer name; for example, sales.contoso.com\computer-01.
+        
+        -- NetBIOS domain name\computer name; for example, sales\computer-01.
+        
+        -- Computer name (host name): for example, computer-01.
+        
+        If the computer name is not specified, the Get-Warranty cmdlet will retrieve warranty information for the computer on which it was run.
+        .PARAMETER Latest
+        Filter out all but the most recent warranty.
+        .EXAMPLE
+        Get-warranty -Computer test-machine-01
+    #>
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory=$false,
+               ValueFromPipeline=$True,
+               ValueFromPipelineByPropertyName=$True)]
+    [Alias('Name')]
+    [String]$Computer=$env:COMPUTERNAME,
+    [Parameter(Mandatory=$false)]
+    [Switch]$Latest
+)
+$apiKey = '1adecee8a60444738f280aad1cd87d0e'
+
+foreach($Comp in $Computer) {
+    $computerIsUp = Test-Connection $Comp -Count 1 -Quiet
+    if (!$computerIsUp) {
+        Write-Error "$Comp is down."
+    } else {
+        # Get Service Tag and Model of target computer.
+        $bios       = gwmi Win32_SystemEnclosure -ComputerName $Comp
+        $system     = gwmi Win32_ComputerSystem -ComputerName $Comp
+        $serviceTag = $bios.SerialNumber
+        $compName   = $bios.__SERVER
+        $model      = $system.Model
+        $manuf      = $system.Manufacturer
+
+        if (!($manuf -match 'Dell')) {
+            Write-Error "Computer not manufactured by Dell. Can't get warranty information."
+        } else {
+            # Get warranty information from Dell's website.
+            $url        = "https://api.dell.com/support/v2/assetinfo/warranty/tags?svctags=${serviceTag}&apikey=${apiKey}"
+            $req        = Invoke-RestMethod -URI $url -Method GET
+            $warranties = $req.getassetwarrantyresponse.getassetwarrantyresult.response.dellasset.warranties.warranty
+            $dellasset  = $req.getassetwarrantyresponse.getassetwarrantyresult.response.dellasset
+
+            # If the $Latest paramater is given, filter out all but the most recent warranty.
+            if ($Latest) {
+                $latestWarranty = $warranties[0]
+                foreach ($warranty in $warranties) {	
+                    if ((Get-Date $warranty.enddate) -gt (Get-Date $latestWarranty.enddate)) {
+                        $latestWarranty = $warranty
+                    }
+                }
+                $warranties = $latestWarranty
+            }
+
+            # Construct and write output object.
+            foreach($warranty in $warranties) {
+                $age = [datetime]::ParseExact($dellasset.shipdate,"yyyy-MM-ddTHH:mm:ss",$null)
+                $age = "{0:N2}" -f (([datetime]::now - $age).days / 365)
+                
+                $output = New-Object -Type PSCustomObject
+                Add-Member -MemberType NoteProperty -Name 'ComputerName' -Value $comp -InputObject $output
```

Comment:
Can you change the value variable to $compName instead and resubmit? That way, you get the computer's actual name as it knows it, rather than the argument passed in by the user which could be a DNS name/alias or something like 'localhost'.